### PR TITLE
[Fix] Fix `infer_onnx.py` Data Loading Bug

### PR DESCRIPTION
**Pull request by Bobholamovic**
修复了`infer_onnx.py`对输入数据处理的bug。

### CODE REVIEW

**Review by zhengya01 — APPROVED** [review]

**Review by juncaipeng — APPROVED** [review]
LGTM

**Comment by dFuong** [comment]
image

I meet an trouble 
Please help me ! @juncaipeng @zhengya01

**Comment by Bobholamovic** [comment]
> image
> 
> I meet an trouble Please help me ! @juncaipeng @zhengya01

What is likely to cause this error is that the `result` passed to `get_pseudo_color_map` seems not to be a properly shaped indexed image (which is what you got after applying argmax), but rather, an image with multiple channels (e.g. the logits). Please check that the output of your network has the shape [H, W].

**Comment by dFuong** [comment]
> 

image
There are input and output model

**Comment by Bobholamovic** [comment]
Please ensure that argmax is applied to the logits to get the final sementation map of shape [B, H, W]. You can specify the output_op when exporting the model.

**Comment by dFuong** [comment]
> 

yep, I applied argmax but output wasn't expected
image

**Comment by Bobholamovic** [comment]
> > 
> 
> yep, I applied argmax but output wasn't expected image

The shape of the right image does not look correct. How did you argmax the network output?

**Comment by dFuong** [comment]
image

I add `result = np.argmax(result, axis=-1)` in line 42

**Comment by Bobholamovic** [comment]
> > 
> 
> image There are input and output model

According to this, the channel dimension is at dimension 1, so you should write `result = np.argmax(result, axis=1)`.

**Comment by dFuong** [comment]
yepp, I understand. Thanks your relpy.

I have another question "How to get bounding box object ?" 
@Bobholamovic

**Comment by Bobholamovic** [comment]
> dFuong

AFAIK, PaddleSeg does not provide a model that outputs bounding boxes. All models provided by PaddleSeg produce masks during inference, which is common in image segmentation tasks. 

I suggest that you should open an issue and give a full description on your question. Anyway, this is a closed pull request and we are not supposed to have lengthy discussions here.
